content/blog/2025-11-05-valkey-helm-chart/index.md

```diff
@@ -0,0 +1,141 @@
+
+
+### Essential capabilities in the Valkey Helm Chart.
+
+This chart focuses on the basics. It gives you a clean default and a few knobs you actually need.
+
+* **Small cache for one service.** One Valkey pod, no persistence, ClusterIP service. Fast to start. Good for stateless caches. See the install steps in the README: [https://github.com/valkey-io/valkey-helm](https://github.com/valkey-io/valkey-helm).
+* **Read-heavy traffic.** One primary with two replicas. Point reads at a separate service if you prefer. Writes still go to the primary. Configure replicas in `values.yaml` and upgrade with Helm.
+* **Simple durability.** Turn on PVCs. Pick a storage class your cluster supports. Back up PVCs with your platform’s snapshot tools while chart-level hooks are being designed.
+* **Safe deploys.** Pin your chart version and Valkey image tag in `values.yaml`. Use `helm diff` in CI. If probes fail after an upgrade, roll back immediately with Helm.
+
+## How to migrate from Bitnami to the Valkey Helm chart (in-place).
+
+**This path has downtime.** Plan a short **maintenance window** and pause writers during the swap.
+
+### 1) Find your release and namespace
+
+```
+helm list --all-namespaces# Example: NAME=valkey-prod  NAMESPACE=acme-valkey
+kubectl get pods,svc,sts,pvc -n acme-valkey
+
+```
+
+### 1) Back up and capture current values
+
+```
+helm get values valkey-prod -n acme-valkey -o yaml > current-values.yaml# Also take an AOF/RDB or storage snapshot
+
+```
+
+### 2) Add the Valkey-helm repo
+
+```
+helm repo add valkey https://valkey.io/valkey-helm/
+helm repo update
+```
+
+### 3) Prepare migration overrides
+
+Match names so the new chart **reuses the same PVCs and Services**. Create `migrate-values.yaml`:
+
+```
+# Set to the names your Bitnami release created
+nameOverride: "<bitnami-short-name>"      # e.g., "valkey"
+fullnameOverride: "<bitnami-full-name>"   # e.g., "valkey-master"
+service:
+name: "<existing-service-name>"         # e.g., "valkey"
+port: 6379
+persistence:
+enabled: true
+size: "<existing-pvc-size>"             # e.g., "8Gi"
+storageClass: "<existing-class-or-null>"
+
+```
+
+Tip: confirm with
+
+```
+kubectl get sts,svc,pvc -n acme-valkey -o wide
+```
+
+### 4) Dry run (recommended)
+
+```
+helm upgrade valkey-prod valkey/valkey \
+  -n acme-valkey \
+  -f current-values.yaml \
+  -f migrate-values.yaml \
+  --dry-run
+```
+
+### 5) Enter maintenance window (pause writes)
+
+### 6) Perform the in-place swap
+
+```
+helm upgrade valkey-prod valkey/valkey \
+  -n acme-valkey \
+  -f current-values.yaml \
+  -f migrate-values.yaml \
+  --atomic --wait --timeout 10m
```

Comment:
what `wait` and `timeout` mean? I think the reader should know if these are opinionated arguments or required?